### ISSUE TITLE
[BC Idea]: Shopify: Save and expose Product Creation Errors 

### ISSUE THREAD

**Issue by eivindgiske** [microsoft/ALAppExtensions#26610]
### BC Idea Link

https://experience.dynamics.com/ideas/idea/?ideaid=9f0096c1-f9c4-ee11-92bd-000d3a058698

### Description

At the moment, the Shopify Connector for Dynamics 365 Business Central does not have error handling on Product Sync (Creation in BC). As an end user, BC Items may not be created, and the error message is only available using VS Code debugger.

The Shopify Products page should have an Error Message field, like Shopify Orders, so that users are able to handle errors in product creation without being dependent on BC Developers.

The hidden error message is stored in variable ErrMsg: Text in codeunit 30185 "Shpfy Sync Products"

### I will provide the implementation for this BC Idea

- [X] I will provide the implementation for this BC Idea
Internal work item: [AB#540964](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/540964)

**Comment by JesperSchulz** [comment]
@AndreiPanko / @onbuyuka, could you triage this idea, please?

**Comment by AndreiPanko** [comment]
@eivindgiske 
I understand your scenario is about converting imported Shopify Product to Item in Business Central. We haven't heard any feedback about that (except number series to run out of numbers).  Would be interesting to learn where do you see errors?

Also there is no action to create item from product (variant), only mapping. So once user sees the error and solves the underlying bloker (changes no series), what is the action for them to take?

**Comment by eivindgiske** [comment]
Hi @AndreiPanko.

We have implented BC onto an existing Shopify environment, and have used a setup where Shopify (and PIM) are Master for Product information.

Using this setup in BC Shopify Shop, the following business logic will execute.
- Sync Item: From Shopify
- Auto Create Unknown Items: TRUE

Report _"Shpfy Sync Products" will execute Codeunit _"Shpfy Product Import"_ which in turn will evaluate whether an item exists or not, and execute _Codeunit "Shpfy Create Item"_ if an item is unknown to BC. You are correct, there is no Action on the Page Shopify Products to do this manually, but it runs on the product sync.

There are integration events on Codeunit "Shpfy Create Item" that we use to control other aspects of product information. Things might fail for a number of reasons other than No. Series.

I cannot see why a partial error handling has been implemented, but is not stored on the Shopify Products record, and we urgently need this in place.

**Comment by AndreiPanko** [comment]
> 
> There are integration events on Codeunit "Shpfy Create Item" that we use to control other aspects of product information. Things might fail for a number of reasons other than No. Series.
> 

Good point about extensibility. @JesperSchulz  what are the next steps?

**Comment by JesperSchulz** [comment]
We approve this issue and wait for a PR 🙂

**Comment by onbuyuka** [comment]
Shopify is moved to [BcApps ](https://github.com/microsoft/BCApps/tree/main/src/Apps/W1/Shopify) repo, closing this due to no activity.
